public/app/api/clients/folder/v1beta1/index.ts

```diff
@@ -1,8 +1,195 @@
+import { useGetFolderQuery as useGetFolderQueryLegacy } from '../../../../features/browse-dashboards/api/browseDashboardsAPI';
+import { PAGE_SIZE } from '../../../../features/browse-dashboards/api/services';
+import { refetchChildren } from '../../../../features/browse-dashboards/state/actions';
+import { GENERAL_FOLDER_UID } from '../../../../features/search/constants';
+import { useGetDisplayMappingQuery } from '../../iam/v0alpha1';
+
 import { generatedAPI } from './endpoints.gen';
+import { rootFolder, sharedWithMeFolder } from './virtualFolders';
 
-export const folderAPIv1beta1 = generatedAPI.enhanceEndpoints({});
+export const folderAPIv1beta1 = generatedAPI.enhanceEndpoints({
+  endpoints: {
+    deleteFolder: {
+      onQueryStarted: (args, { queryFulfilled, dispatch }) => {
+        // Refetch for anything using the old API.
+        queryFulfilled.then(() => {
+          dispatch(
+            refetchChildren({
+              parentUID: GENERAL_FOLDER_UID,
+              pageSize: PAGE_SIZE,
+            })
+          );
+        });
+      },
+    },
+  },
+});
 
+const { useGetFolderParentsQuery, useGetFolderAccessQuery } = folderAPIv1beta1;
 export const { useGetFolderQuery } = folderAPIv1beta1;
 
+function getFolderUrl(uid: string, title: string): string {
+  // mimics https://github.com/grafana/grafana/blob/79fe8a9902335c7a28af30e467b904a4ccfac503/pkg/services/dashboards/models.go#L188
+  // Not the same slugify as on the backend https://github.com/grafana/grafana/blob/aac66e91198004bc044754105e18bfff8fbfd383/pkg/infra/slugify/slugify.go#L86
+  // Probably does not matter as it seems to be only for better human readability.
+  const slug = kbn.slugifyForUrl(title);
+  return `${config.appSubUrl}/dashboards/f/${uid}/${slug}`;
+}
+
+/**
+ * A proxy function that uses either legacy folder client or the new app platform APIs to get the data in the same
+ * format of a FolderDTO object. As the schema isn't the same, using the app platform needs multiple different calls
+ * which are then stitched together.
+ * @param uid
+ */
+export function useGetFolderQueryFacade(uid?: string) {
+  if (config.featureToggles.foldersAppPlatformAPI) {
+    const isVirtualFolder = uid && [GENERAL_FOLDER_UID, config.sharedWithMeFolderUID].includes(uid);
+    const params = isVirtualFolder || !uid ? skipToken : { name: uid };
+
+    let result = useGetFolderQuery(params);
+
+    const resultParents = useGetFolderParentsQuery(params);
+    const resultAccess = useGetFolderAccessQuery(params);
+
+    const keys: string[] = result.data
+      ? [
+          result.data.metadata.annotations?.[AnnoKeyUpdatedBy],
+          result.data.metadata.annotations?.[AnnoKeyCreatedBy],
+        ].filter((v) => v !== undefined)
+      : [];
+
+    const needsUserData = !isVirtualFolder && Boolean(keys.length);
+    const resultUserDisplay = useGetDisplayMappingQuery(needsUserData ? { key: keys } : skipToken);
+
+    if (isVirtualFolder) {
```

Comment:
@Clarity-89 So this is the take on the virtual folders just simulating them here instead of the backend which seems like a preference from the backend team, compared to putting it somewhere in the unified storage path.